testing/repository_client.go

```diff
@@ -0,0 +1,85 @@
+package vtesting
+
+import (
+	"context"
+    "regexp"
+	"github.com/Shopify/voucher/repository"
+)
+const (
+    repoRegex = "^(?:https?|git)(?:://|@)(?P<vcs>[^/:]+)[/:](?P<org>[^/.]+)/(?P<name>[^/.]+)(?:.git)?"
+    commitRegex = "^(?:https?|git)(?:://|@)(?P<vcs>[^/:]+)[/:](?P<org>[^/.]+)/(?P<name>[^/.]+)/(?:commit)/(?P<hash>[^/.]+)"
+)
+
+type repositoryclient struct {
+    repos map[string]Repository
+}
+
+//NewClient creates a new repository client
+func NewClient() (RepositoryClient, error) {
+	return &repositoryclient{}, nil
+}
+
+//AddRepository adds to repository to the repository client
+func (c *repositoryclient) AddRepository(org repository.Organization, name string, commits map[string]repository.Commit, branches map[string]repository.Branch) {
+    if c.repos == nil {
+        c.repos = make(map[string]Repository)
+    }
+    c.repos[org.Name] = Repository{org, name, commits, branches}
+}
+
+//GetCommitInfo creates commit info with provided
+func (c *repositoryclient) GetCommit(ctx context.Context, commitURL string) (repository.Commit, error) {
+    re := regexp.MustCompile(commitRegex)
+    res := re.FindStringSubmatch(commitURL)
+    org_name := res[2]
+    repo := c.repos[org_name]
+    commit := repo.Commits[commitURL]
+	return commit, nil
+}
+
+//GetOrganization creates organization
+func (c *repositoryclient) GetOrganization(ctx context.Context, uri string) (repository.Organization, error) {
+    var org repository.Organization
+    re := regexp.MustCompile(repoRegex)
+    org_name := re.FindStringSubmatch(uri)[2]
+    if repo, exists := c.repos[org_name]; exists{
+        org = repo.Org
+    }
+	return org, nil
+}
+
+func (c *repositoryclient) GetDefaultBranch(ctx context.Context, commit repository.Commit) (repository.Branch, error) {
+	var branch repository.Branch
+    re := regexp.MustCompile(repoRegex)
+    org_name := re.FindStringSubmatch(commit.URL)[2]
+    
+    branches := c.repos[org_name].Branches
+    for _, br := range branches {
+        commits := br.CommitRefs
+        for _, c := range commits {
+            if c.URL == commit.URL {
+                branch = br
+                break
+            }
+        }
+    }
+	return branch, nil
+}
+
+
+func (c *repositoryclient) GetBranch(ctx context.Context, commit repository.Commit) (repository.Branch, error) {
+    var branch repository.Branch
+    re := regexp.MustCompile(repoRegex)
+    org_name := re.FindStringSubmatch(commit.URL)[2]
+    
+    master := c.repos[org_name].Branches["master"]
+    commits := master.CommitRefs
+    for i:=0; i < len(commits); i++ {
```

Comment:
Like the other functions, you can interate through slices using `range`:

```suggestion
    for i, commit := range branch.CommitRefs {
```

Reply:
Done